src/js/panels.js

```diff
@@ -322,76 +322,126 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                tooltipAtSelect: ["yes-tooltipAtSelect", "no-tooltipAtSelect"]
             },
             choiceInput: {
-                tooltip: ["speakText-yes-tooltip", "speakText-no-tooltip"],
-                tooltipAtSelect: ["speakText-yes-tooltipAtSelect", "speakText-no-tooltipAtSelect"]
+                tooltip: ["yes-tooltip", "no-tooltip"],
+                tooltipAtSelect: ["yes-tooltipAtSelect", "no-tooltipAtSelect"]
             }
         },
         stringArrayIndex: {
-            choice: ["speakText-yes", "speakText-no"]
+            choice: ["yes", "no"]
         },
         selectors: {
-            choiceRow: ".gpiic-fd-speakText-choiceRow",
-            choiceLabel: ".gpiic-fd-speakText-choice-label",
-            choiceInput: ".gpiic-fd-speakText-choiceInput",
-            instructions: ".gpiic-fd-speakText-instructions"
+            choiceRow: ".gpiic-fd-yesNo-choiceRow",
+            choiceLabel: ".gpiic-fd-yesNo-choiceLabel",
+            choiceInput: ".gpiic-fd-yesNo-choiceInput",
+            instructions: ".gpiic-fd-yesNo-instructions"
         },
         controlValues: {
             choice: ["yes", "no"]
         },
         repeatingSelectors: ["choiceRow"],
-        protoTree: {
-            instructions: {messagekey: "speakTextInstructions"},
+        invokers: {
+            produceTree: {
+                funcName: "gpii.firstDiscovery.panel.yesNo.produceTree",
+                args: "{that}"
+            }
+        }
+    });
+
+    gpii.firstDiscovery.panel.yesNo.produceTree = function () {
+        // Make sure each derived panel using yesNo grade has a unique
+        // selectID, the name used for inputs.
+        var selectID = fluid.allocateGuid();
+        var protoTree = {
+            instructions: {messagekey: "instructions"},
             expander: {
                 type: "fluid.renderer.selection.inputs",
                 rowID: "choiceRow",
                 labelID: "choiceLabel",
                 inputID: "choiceInput",
-                selectID: "choice-radio",
+                selectID: selectID,
                 tree: {
                     optionnames: "${{that}.msgLookup.choice}",
                     optionlist: "${{that}.options.controlValues.choice}",
-                    selection: "${speakChoice}"
+                    selection: "${choice}"
+                }
+            }
+        };
+        return protoTree;
+    };
+
+    /*
+     * Text to speech panel
+     */
+    fluid.defaults("gpii.firstDiscovery.panel.speakText", {
+        gradeNames: ["gpii.firstDiscovery.panel.yesNo", "autoInit"],
+        preferenceMap: {
+            "gpii.firstDiscovery.speak": {
+                "model.speak": "default"
+            }
+        },
+        modelRelay: {
+            source: "{that}.model.choice",
+            target: "{that}.model.speak",
+            // Setup the backward restriction to prevent the component instantiation writes back to
```

Comment:
The modelRelay for speakText and onScreenKeyboard seem to be the same except for the target. Maybe we should just map them all to the same model path (e.g. model.value) in the preferenceMap so that you could reuse the modelRelay across components too.